tradssat/tmpl/file.py

```diff
@@ -58,7 +58,22 @@ def get_var_type(self, var, sect=None):
     def get_var_lims(self, var, sect=None):
         return self.get_var(var, sect).lims
 
-    def get_var_spc(self, var, sect=None):
+    def get_var_spc(self, var, sect=None, **kwargs):
```

Comment:
Yes, It's a much clean way! 

----------------------------

Anyway, we still need the `header` to process the outputs files. 

It hurts the common rule a little:

`whenever you override a method, the interface of the new method should be
the same as the old.  It should take the same parameters, return the same
type, and obey the same preconditions and postconditions`

----------------------------

The best implementation should be when the `spc` is first assigned. Until now, I do not know how to do it.

Reply:
> Would it be possible to implement this instead by overriding get_var_spc in the subclass OutFile?

Done!

Reply:
Looks fine; in that case, it wouldn't hurt to add **kwargs to all the implementations of get_var_spc. Regarding changing this when `spc` is assigned, I think you are right. But it would require a large rewrite of the internal code, so I may get around to it in a few months! If this works for now, I think we should just accept this pull request (ideally with overridden get_var_spc in OutFile and new tests, if you are able to add those) until something better can be done.
Thanks!